Sources/Authgear.swift

```diff
@@ -531,15 +539,15 @@ public class Authgear: NSObject {
                     jwt: nil
                 )
 
-                handler?(self.persistSession(tokenResponse))
+                handler?(self.persistSession(tokenResponse, reason: .foundToken))
             } catch {
                 if let error = error as? AuthAPIClientError,
-                    case let .oidcError(oidcError) = error,
-                    oidcError.error == "invalid_grant" {
+                   case let .oidcError(oidcError) = error,
+                   oidcError.error == "invalid_grant" {
                     return DispatchQueue.main.async {
-                        self.setSessionState(.noSession)
+                        self.setSessionState(.noSession, reason: .expired)
```

Comment:
Should we remove line 548?
`authgearSessionStateDidChange` would be called twice.